app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -65,7 +72,26 @@ def send_confirmation_code
     # user's session. Re-sending the confirmation code doesn't generate a new one.
     self.confirmation_code = generate_confirmation_code unless confirmation_code
 
-    SmsSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
+    if current_otp_delivery_method == :voice
```

Comment:
I think this is fine the way it is. I think this line of code perfectly captures the intended behavior: if the user selected voice, send voice OTP, otherwise send SMS. If we really want to just return error, that is fine as well. 

I think changing the logic of `sms_enabled?` to the above is actually less clear.